src/System.IO.UnmanagedMemoryStream/src/System/IO/UnmanagedMemoryStream.cs

```diff
@@ -429,34 +421,24 @@ public override long Position
             [System.Security.SecurityCritical]  // auto-generated_required
             get
             {
-                if (_buffer != null)
-                {
-                    throw new NotSupportedException(Strings.NotSupported_UmsSafeBuffer);
-                }
+                if (_buffer != null) throw new NotSupportedException(Strings.NotSupported_UmsSafeBuffer);
+                if (!_isOpen) throw __Error.GetStreamIsClosed();
 
                 // Use a temp to avoid a race
                 long pos = Interlocked.Read(ref _position);
-                if (pos > _capacity)
-                    throw new IndexOutOfRangeException(Strings.IndexOutOfRange_UMSPosition);
+                if (pos > _capacity) throw new IndexOutOfRangeException(Strings.IndexOutOfRange_UMSPosition);
                 byte* ptr = _mem + pos;
-                if (!_isOpen) throw __Error.GetStreamIsClosed();
+
                 return ptr;
             }
             [System.Security.SecurityCritical]  // auto-generated_required
             set
             {
-                if (_buffer != null)
-                    throw new NotSupportedException(Strings.NotSupported_UmsSafeBuffer);
+                if (_buffer != null) throw new NotSupportedException(Strings.NotSupported_UmsSafeBuffer);
                 if (!_isOpen) throw __Error.GetStreamIsClosed();
+                if (value < _mem) throw new IOException(Strings.IO_SeekBeforeBegin);
 
-                // Note: subtracting pointers returns an Int64.  Working around
-                // to avoid hitting compiler warning CS0652 on this line. 
-                if (new IntPtr(value - _mem).ToInt64() > UnmanagedMemStreamMaxLength)
-                    throw new ArgumentOutOfRangeException("offset", Strings.ArgumentOutOfRange_UnmanagedMemStreamLength);
-                if (value < _mem)
-                    throw new IOException(Strings.IO_SeekBeforeBegin);
-
-                Interlocked.Exchange(ref _position, value - _mem);
+                Seek(value - _mem, SeekOrigin.Begin);
```

Comment:
Regarding duplication and differences, the Position property has additional logic applied for when IntPtr.Size == 4.  Do you know why that logic doesn't also exist in Seek?

Also, if I'm reading this correctly, I think your change introduces the possibility of an exception that wasn't previously there (this might be desirable, but I wanted to call it out, regardless).  PositionPointer could be set to a value such that value - _mem could be larger than Int64.MaxValue, and when cast to a long it'd end up wrapping around to be negative.  Previously that negative value would just be stored into _position in the Interocked.Exchange, but now it'll be passed to Seek, which will compare it against 0 and throw an exception for seeking before the beginning.